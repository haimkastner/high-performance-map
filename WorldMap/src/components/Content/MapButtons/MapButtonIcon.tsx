import React from "react";
import { Box } from "@material-ui/core";

/**
 * the properties for an Icon in one of the map buttons. 
 */
interface Props {
  imageName: string;
  width?: string;
  height?: string;
}

/**
 * Gets the @param props to put on the button's image.
 * @returns the image with the updated css properties.
 */
export class MapButtonIcon extends React.Component<any, Props> {
  render() {
    return (
      <Box>
        <img
          width={this.props.width ?? 25} //if not given, default is 25
          height={this.props.height ?? 25} //if not given, default is 25
          src={require(`./../../../assets/img/${this.props.imageName}`)} //the complete path with the name
        />
      </Box>
    );
  }
}
