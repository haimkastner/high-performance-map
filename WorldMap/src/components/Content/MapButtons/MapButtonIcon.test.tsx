import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MapButtonIcon } from './MapButtonIcon';

describe('MapButtonIcon Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MapButtonIcon imageName={"side-menu-shape-copy-2@2x.png"} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<MapButtonIcon imageName={"side-menu-shape-copy-2@2x.png"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // tslint:disable-next-line: no-empty
  describe('MapButtonIcon Component Logic', () => {});
});
