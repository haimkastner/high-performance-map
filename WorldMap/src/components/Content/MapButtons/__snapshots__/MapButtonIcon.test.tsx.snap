// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapButtonIcon Component matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-4"
>
  <img
    height={25}
    src="side-menu-shape-copy-2@2x.png"
    width={25}
  />
</div>
`;
